test/parser-test.js

```diff
@@ -26,9 +26,13 @@ describe('Parser', () => {
     it('can parse a basic ruleset', () => {
```

Comment:
Abbreviation has the unfortunate side-effect of retaining noise while adding obfuscation.

I was considering de-structuring the methods of the `NodeFactory` to individual identifiers, since they are stateless in the base implementation. This has the downside of being potentially brittle over time (the potential is small, IMO). But, it has the upside of achieving the outcome you recommend without adding obfuscated names into the domain. WDYT?